### PR TITLE
[Docs] Add application runtime

### PR DESCRIPTION
**Pull request by jillnogold**

### CODE REVIEW

File: docs/runtimes/application.ipynb [hunk 1]
```diff
@@ -0,0 +1,103 @@
+{
```
Comment (alonmr) [inline docs/runtimes/application.ipynb:1]: <ol><li>I would remove the <code>(a web application)</code> part from the top or change it to <code>(e.g. a web application)</code></li><li> pull at run-time is not supported yet.</li><li>API gateway can be disabled to allow free access without authentication</li></ol><p>		&gt;The deployment ensures an API Gateway</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/mlrun/mlrun/pull/6036/discussion/#comment-1687746348'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='0'/>

Comment (alonmr) [inline docs/runtimes/application.ipynb:1]: <p>set_internal_app_port was change to set_internal_application_port</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/mlrun/mlrun/pull/6036/discussion/#comment-1687746351'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='1'/>

Comment (alonmr) [inline docs/runtimes/application.ipynb:1]: <p>This feature was cancelled</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/mlrun/mlrun/pull/6036/discussion/#comment-1687746353'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='3'/>

Comment (alonmr) [inline docs/runtimes/application.ipynb:1]: <p>need to change a few things the example is not up to date</p><p>add the following lines:</p><p>application.set_internal_application_port(&lt;add-your-port&gt;)</p><p>application.spec.command = &lt;add-command-string&gt;</p><p>application.spec.args = &lt;add-command-args-list-of-strings&gt;</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/mlrun/mlrun/pull/6036/discussion/#comment-1687746357'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='5'/>

File: docs/runtimes/application.ipynb [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
```
Comment (TomerShor) [inline docs/runtimes/application.ipynb:1]: <blockquote>The application runtime deploys a user container image (for example, a web application), a port, and a command to run the HTTP server. </blockquote><p>This sentence is not accurate, as it doesn't "deploy a port and a command", it deploys a container image that is exposed on a specific port.</p><p> </p><blockquote>or use a pre-built image</blockquote><p>As Alon said, this is not supported.</p><p> </p><blockquote>you can disable it to allow free access without authentication. </blockquote><p>The authentication can be disabled, not the API Gateway. Maybe something like:</p><p>"... is in front of the application and can provide different authentication methods, or none"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/mlrun/mlrun/pull/6036/discussion/#comment-1688678280'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='0'/>

**Comment by jillnogold** [comment]
@alonmr @TomerShor - Thanks!

File: docs/concepts/nuclio-real-time-functions.ipynb [hunk 3]
```diff
@@ -167,11 +167,11 @@
...
     "Create (or update) the API gateway</br>\n",
     "It's crucial to update the defined API Gateway entity with the entity returned from store_api_gateway method. This fills in important fields like host/path, etc.\n",
-    "```\n",
+    "```python\n",
     "my_api_gateway = project.store_api_gateway(my_api_gateway)\n",
     "```\n",
-    "store_api_gateway checks that api_gateway is ready to be invoked before returning it. You can also check if api gateway is in ready state:\n",
-    "```\n",
+    "`store_api_gateway` checks that api_gateway is ready to be invoked before returning it. You can also check if api gateway is in ready state:\n",
+    "```python\n",
```
Comment (TomerShor) [inline docs/concepts/nuclio-real-time-functions.ipynb]: i'm not sure the `python` magic is working in `ipynb` files, it's mostly for `md` files.

File: docs/runtimes/application.ipynb [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "(application)=\n",
+    "# Application runtime\n",
+    "\n",
+    "You can use the {py:meth}`~mlrun.runtimes.ApplicationRuntime` to provide an image that runs on top of your deployed model. \n",
+    "\n",
+    "The application runtime deploys a container container image (for example, a web application) that is exposed on a specific port, and a command to run the HTTP server. The runtime is based on top of Nuclio, and adds the application as a side-car to a Nuclio function while the actual function is a reverse proxy to that application. The application runtime builds the side-car image. You can specify the source code, and pull at build-time. An API Gateway, by default, is in front of the application  and can provide different authentication methods, or none.\n",
```
Comment (TomerShor) [inline docs/runtimes/application.ipynb]:
```suggestion
    "The application runtime deploys a container image (for example, a web application) that is exposed on a specific port, and a command to run the HTTP server. The runtime is based on top of Nuclio, and adds the application as a side-car to a Nuclio function pod while the actual function is a reverse proxy to that application. 
    You can set an existing image to run in the application , or let the application runtime build the side-car image for you, by specifying the source code, and pulling at build-time. 
    An API Gateway, by default, is in front of the application and can provide different authentication methods, or none.\n",
```

**Review by TomerShor — CHANGES_REQUESTED** [review]

**Review by TomerShor — APPROVED** [review]

**Pull request merged**